### PR TITLE
Add support for changes included in Compose 1.2.0

### PR DESCRIPTION
**Pull request by rafaelfrancisco-dev**
# Description

This PR makes changes to accommodate the LazyGrid changes that will come out with Compose 1.2.0. Since some users are already using the preview version on their apps it breaks the current version of compose-material-dialog.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit and screenshot tests pass locally with my changes

### CODE REVIEW

**Comment by vanpra** [comment]
Looks like you've introduced a few wildcard imports. If change those back to full imports the CI should pass and then I can merge this in

**Comment by rafaelfrancisco-dev** [comment]
I replaced the wildcard imports and it built fine in the CI and the functional tests passed. However the screenshot tests failed with an expected size error, any clues as to why ?

**Comment by vanpra** [comment]
It looks like there's been a change in the size of the dialog between the stable version and the alpha meaning the screenshot test no longer match. This may just be a case of recapturing the screenshot tests but I will have to take a look at why the size changed in the first place.

**Comment by rafaelfrancisco-dev** [comment]
Thanks, if anything changes I'll update the PR

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -44,7 +44,7 @@ object Dependencies {
         }
 
         object Compose {
-            const val version = "1.1.1"
+            const val version = "1.2.0-alpha07"
```
Comment (rahulsainani) [inline buildSrc/src/main/kotlin/Dependencies.kt:47]: Shall we update to `beta-01` (or the latest Compose library when the tests pass)?

**Comment by skolson** [comment]
FYI see comments on issue 137. I made them without realizing this PR was already in-flight. I'm +1 for the beta version and for the aplpha11 version of material3.  But be aware of a problem with the month change buttons (described in issue 137 comments) using these newest versions (since alpha08). To get those buttons working needs either a workaround for [Compose issue]( https://issuetracker.google.com/issues/229752147), or wait until [Google patch](https://android-review.googlesource.com/c/platform/frameworks/support/+/2073730/) is included in a release.

**Pull request closed without merging**